apps/src/sites/studio/pages/layouts/_student_information_interstitial.js

```diff
@@ -0,0 +1,59 @@
+import $ from 'jquery';
+import getScriptData from '@cdo/apps/util/getScriptData';
+import analyticsReporter from '@cdo/apps/lib/util/AnalyticsReporter';
+import {EVENTS} from '@cdo/apps/lib/util/AnalyticsConstants';
+
+const retrieveInfoForCap = getScriptData('retrieveInfoForCap');
+const userId = getScriptData('userId');
+const inSection = getScriptData('inSection');
+
+$(document).ready(function () {
+  const pathName = window.location.pathname;
+
+  if (pathName !== '/lti/v1/authenticate') {
+    $('#student-information-modal').modal('show');
+    retrieveInfoForCap &&
+      analyticsReporter.sendEvent(EVENTS.CAP_STATE_FORM_SHOW, {
+        user_id: userId,
+        in_section: inSection,
+      });
+  }
+
+  const checkInputs = () => {
+    const ageValue = $('#user_age').val();
+    const stateValue = $('#user_us_state').val();
+    $('.age-required').toggle(!ageValue);
+    $('.state-required').toggle(!stateValue);
+    console.log('Age Value: ' + ageValue);
+    console.log('Age Value: ' + stateValue);
+    $('#edit_user #submit-btn').prop(
+      'disabled',
+      ageValue === '' || stateValue === ''
+    );
+  };
+
+  $('#edit_user select').on('change', checkInputs);
+
+  $('#edit_user').submit(function (event) {
+    event.preventDefault($(this).serialize());
+    const stateValue = $('#user_us_state').val();
+    if (!stateValue && retrieveInfoForCap)
+      analyticsReporter.sendEvent(EVENTS.CAP_STATE_FORM_PROVIDED, {
+        user_id: userId,
+        in_section: inSection,
+        us_state: stateValue,
+      });
+    $.ajax({
+      type: 'POST',
+      url: $(this).attr('action') + '/set_student_information',
+      data: $(this).serialize(),
+      dataType: 'json',
+      success: function (data) {
+        retrieveInfoForCap
+          ? location.reload()
+          : $('#student-information-modal').modal('hide');
+      },
+    });
+  });
```

Comment:
I'd suggest using the [`remote: true`](https://tadhao.medium.com/rails-way-of-ajax-with-remote-true-7427089a2a57) approach instead. Please check: https://github.com/code-dot-org/code-dot-org/commit/f6530a6561b48208bbcba183b43e3e31df8be146